src/org/zaproxy/zap/extension/fuzz/payloads/generator/RegexPayloadGenerator.java

```diff
@@ -234,6 +245,47 @@ public void close() {
 
     }
 
+    private static class RandomRegexIterator implements ResettableAutoCloseableIterator<DefaultPayload> {
```

Comment:
Please change `RegexIterator` instead, the implementation is pretty much the same (it just requires two ifs, one in the constructor and the other in `reset`).